xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/mockData.js

```diff
@@ -54,6 +60,19 @@ define({
     layouts: ["table", "cards"],
     defaultLayout: "table",
 
+    layoutDescriptors: [
+      {
+        id: "table",
+        name: "Table",
+        icon: "fa fa-table",
```

Comment:
Do we have to use a macro for that? Is there not a javascript method that already creates the icon from this metadata (somewhere in the `XWiki` object for instance)?

Reply:
I don't think there is any JavaScript code you can reuse. Here's what the suggest input does https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/suggest/xwiki.selectize.js#L51 .

Reply:
Okay I see.

I think that would be a good thing to have some XWiki utils methods like this one directly available in JS, so that it avoid code duplication, and adds consistency by avoiding having different implementations of the same thing :)